spring-cloud-gcp-security-iap/src/main/java/org/springframework/cloud/gcp/security/iap/AudienceValidator.java

```diff
@@ -55,16 +57,17 @@ public AudienceValidator(AudienceProvider audienceProvider) {
 
 	@Override
 	public OAuth2TokenValidatorResult validate(Jwt t) {
-		if (t.getAudience() != null && t.getAudience().contains(this.audience)) {
-			return OAuth2TokenValidatorResult.success();
-		}
-		else {
-			if (LOGGER.isWarnEnabled()) {
-				LOGGER.warn(String.format(
-						"Expected audience %s did not match token audience %s", this.audience, t.getAudience()));
+		String[] audiences = StringUtils.trimArrayElements(StringUtils.commaDelimitedListToStringArray(this.audience));
```

Comment:
Could you make `audiences` an instance variable, and move this logic into `afterPropertiesSet()`? This way, the audience splitting will only be done once.

Reply:
Got it!